include/exadg/convection_diffusion/spatial_discretization/operator.cpp

```diff
@@ -820,14 +820,22 @@ template<int dim, typename Number>
 void
 Operator<dim, Number>::move_grid(double const & time) const
 {
-  grid_motion->update(time, false);
+  // Driver::ale_update() is responsible for the preconditioner update
+  grid_motion->update(
+    time,
+    false /* print_solver_info */,
+    dealii::numbers::invalid_unsigned_int /* time_step_number used for preconditioner update */);
```

Comment:
The problem here is that the function `move_grid()` is public and can be used in a different context that our current `Driver`. In this sense, `Operator` is a low level module, whose implementation should not depend on our current high-level implementation in `Driver`. As a consequence, there should actually also not appear a comment on `Driver` in this class (in an ideal world).

I think the origin of this problem is that the two functions `move_grid()` and `move_grid_and_update_dependent_data_structures()` are a sub-optimal design. I think we should remove `move_grid()` and let the time integrator directly operate on `grid_motion`. The function `move_grid_and_update_dependent_data_structures()` needs to be changed to `update_spatial_discretization_after_grid_motion()` (i.e. the first part `grid_motion->update()` should no longer be part of that function). This way, there is only one possibility to move the grid in the future.

In order to proceed with the present PR, I suggest that I could address this in the future once the present PR is merged.

Reply:
I created issue #412 and will therefore resolve the conversation here